weave

```diff
@@ -655,6 +655,10 @@ setup_router_iface_bridge() {
         echo "Perhaps you are running the docker daemon with container networking disabled (-b=none)." >&2
         return 1
     fi
+    # No-op if already attached
+    if netnsenter ip link show $CONTAINER_IFNAME >/dev/null 2>&1 ; then
+        return 0
+    fi
```

Comment:
Well, that exposes a deeper issue.  Using fdp, the `weave` script behaves differently depending on whether you pass a `--password` argument, and we deliberately hide that information from the command-line.

It is sort-of possible to detect which path we went through, from `--datapath weave --iface weave2`, but that's ugly and twisted.